source/includes/fundamentals/code-snippets/CRUD/retrieve.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func main() {
+	var uri string
+	if uri = os.Getenv("MONGODB_URI"); uri == "" {
+		log.Fatal("You must set your `MONGODB_URI' environmental variable. See\n\t https://docs.mongodb.com/drivers/go/current/usage-examples/")
+	}
+
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			panic(err)
+		}
+	}()
+
+	// begin insert docs
+	coll := client.Database("tea").Collection("ratings")
+	docs := []interface{}{
+		bson.D{{"name", "Masala"}, {"rating", 10}},
+		bson.D{{"name", "Earl Grey"}, {"rating", 5}},
+		bson.D{{"name", "Masala"}, {"rating", 7}},
+		bson.D{{"name", "Earl Grey"}, {"rating", 9}},
+	}
+
+	result, err := coll.InsertMany(context.TODO(), docs)
+	if err != nil {
+		panic(err)
+	}
+	// end insert docs
+
+	fmt.Printf("%d documents inserted with IDs:\n", len(result.InsertedIDs))
+	for _, id := range result.InsertedIDs {
+		fmt.Printf("\t%s\n", id)
+	}
+
+	// begin find docs
+	filter := bson.D{
+		{"$and",
+			bson.A{
+				bson.D{{"rating", bson.D{{"$gt", 5}}}},
+				bson.D{{"rating", bson.D{{"$lt", 10}}}},
+			}},
+	}
+
+	findCursor, findErr := coll.Find(context.TODO(), filter)
+	if findErr != nil {
+		panic(findErr)
+	}
+
+	var findResults []bson.D
```

Comment:
Should `findResults` and `aggResults` be the same type for consistency?

This may be worth noting in the [Golang Driver Code Example Patterns and Style Reference](https://docs.google.com/document/d/1FtUKxOaEmdFf4d3HcXAVIGed2awuwIWdkWc0YQIwPvk/edit#). This [slack thread](https://mongodb.slack.com/archives/C028XP912BF/p1628097761003300) suggested `bson.M` for output.